src/utils/git.ts

```diff
@@ -0,0 +1,47 @@
+import * as vscode from "vscode";
+import retry from "async-retry";
+
+const DOT_GIT_SUFFIX_LENGTH = 4;
+
+export function getGitData() {
+  const extensions = vscode.extensions;
+  let normalizedRepositoryName;
+  let currentBranch;
+
+  if (extensions) {
+    const gitExtension = extensions.getExtension("vscode.git")?.exports;
+
+    const api = gitExtension.getAPI(1);
+    const repository = api.repositories[0];
+
+    if (repository) {
+      const head = repository.state.HEAD;
+      currentBranch = head.name;
+      const repositoryName = repository.repository.remotes[0].fetchUrl;
+      normalizedRepositoryName = repositoryName.endsWith(".git")
+        ? repositoryName?.slice(0, -DOT_GIT_SUFFIX_LENGTH)
+        : repositoryName;
+    }
+  }
+
+  return { repository: normalizedRepositoryName, currentBranch };
+}
+export async function getCurrentBranchWithRetry() {
+  return await retry(
+    () => {
+      const result = getGitData();
+      if (!result.currentBranch) {
+        throw new Error("couldn't find git current branch");
+      } else {
+        return result.currentBranch;
+      }
+    },
+    {
+      factor: 1,
+      minTimeout: 300,
+      maxTimeout: 300,
+      randomize: false,
+      retries: 8,
+    }
+  );
+}
```

Comment:
why the retry? why 8 retries?

Reply:
you can see [here](https://github.com/env0/vscode-env0/pull/35#discussion_r1245463829) the full discussion why we need to retry.
i updated the retry to wait up to 9 sec f0b3fe6b989a32de22e1a3be8e2c89e0fb971027.